src/avl_tree.jl

```diff
@@ -0,0 +1,259 @@
+# it has unique keys
+# leftChild has keys which are less than the node
+# rightChild has keys which are greater than the node
+# height stores the height of the subtree.
+mutable struct AVLTreeNode{K}
+    height::Int8
+    leftChild::Union{AVLTreeNode{K}, Nothing}
+    rightChild::Union{AVLTreeNode{K}, Nothing}
+    data::K
+
+    AVLTreeNode{K}(d::K) where K = new{K}(1, nothing, nothing, d)
+end
+
+AVLTreeNode(d) = AVLTreeNode{Any}(d)
+
+AVLTreeNode_or_null{T} = Union{AVLTreeNode{T}, Nothing}
+
+mutable struct AVLTree{T}
+    root::AVLTreeNode_or_null{T}
+    count::Int
+
+    AVLTree{T}() where T = new{T}(nothing, 0)
+end
+
+AVLTree() = AVLTree{Any}()
+
+Base.length(tree::AVLTree) = tree.count
+
+get_height(node::Union{AVLTreeNode, Nothing}) = (node == nothing) ? 0 : node.height
+
+# balance is the difference of height between leftChild and rightChild of a node.
+function get_balance(node::Union{AVLTreeNode, Nothing})
+    if node == nothing
+        return 0
+    else
+        return get_height(node.leftChild) - get_height(node.rightChild)
+    end
+end
+
+fix_height(node::AVLTreeNode) = 1 + max(get_height(node.leftChild), get_height(node.rightChild))
```

Comment:
What does this do?
It doesn't fix the hight since it doesn't mutate.

Perhaps it should be called:
`compute_hight` and have a comment that it computes the hight rather than using the variable?

or make it should be called:
`recompute_height!(node)` and it should update the hight in place?